metals/src/main/scala/scala/meta/internal/troubleshoot/ProblemResolver.scala

```diff
@@ -0,0 +1,161 @@
+package scala.meta.internal.troubleshoot
+
+import scala.collection.mutable.ListBuffer
+
+import scala.meta.internal.metals.Messages
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ScalaTarget
+import scala.meta.internal.metals.ScalaVersions
+import scala.meta.io.AbsolutePath
+
+class ProblemResolver(
+    workspace: AbsolutePath,
+    isUnsupportedBloopVersion: () => Boolean,
+    bloopVersion: () => Option[String],
+    isClientCommandSupported: Boolean
+) {
+
+  def recommendation(scala: ScalaTarget): String = {
+    findProblem(scala)
+      .map(_.message)
+      .getOrElse("")
+  }
+
+  def problemSummary(allTargets: List[ScalaTarget]): Option[String] = {
+
+    val unsupportedVersions = ListBuffer[String]()
+    val deprecatedVersions = ListBuffer[String]()
+    val futureVersions = ListBuffer[String]()
+    var misconfiguredProjects = 0
+    var unsupportedSbt = false
+    var deprecatedSbt = false
+    var futureSbt = false
+
+    for {
+      target <- allTargets
+      issue <- findProblem(target)
+    } yield {
+      issue match {
+        case UnsupportedScalaVersion(version) => unsupportedVersions += version
+        case DeprecatedScalaVersion(version) => deprecatedVersions += version
+        case FutureScalaVersion(version) => futureVersions += version
+        case _: SemanticDBDisabled => misconfiguredProjects += 1
+        case _: MissingSourceRoot => misconfiguredProjects += 1
+        case UnsupportedSbtVersion => unsupportedSbt = true
+        case DeprecatedSbtVersion => deprecatedSbt = true
```

Comment:
For some of these, it seems like we have an extra middle step that isn't really necessary. For example for these, we find the issue, and if it's one of these two, we set it to true, then down below we check if it's true and create a Message. Couldn't we just skip the middle step and instead here instead of returning a Boolean return `Some(message)` or None right away and then there is no need to check for it down below?

Reply:
I think it's easier to understand the way it is. First we check what problems we have and only then translate to the messages. Otherwise we will have to mix those two steps, but I do understand what you mean and I can change it if you feel like it will be better.

Reply:
I get what you mean. For consistency then feel free to just leave it as is.